File: backend/services/supabase_client.py
import os
from supabase import create_client, Client
from NEXO_CORE import config

# Las claves deben estar en el .env env
SUPABASE_URL = os.getenv("SUPABASE_URL")
SUPABASE_SERVICE_KEY = os.getenv("SUPABASE_SERVICE_KEY")

_client = None

def get_supabase() -> Client:
    """Retorna un cliente singleton de Supabase."""
    global _client
    if _client is None:
        if not SUPABASE_URL or not SUPABASE_SERVICE_KEY:
            raise ValueError("SUPABASE_URL o SUPABASE_SERVICE_KEY no configurados en .env")
        _client = create_client(SUPABASE_URL, SUPABASE_SERVICE_KEY)
    return _client

File: supabase/functions/_shared/supabaseClient.ts
import { createClient } from "https://esm.sh/@supabase/supabase-js@2.7.1"

export const getSupabaseClient = (serviceRole = false) => {
  return createClient(
    Deno.env.get('SUPABASE_URL') ?? '',
    serviceRole 
      ? Deno.env.get('SUPABASE_SERVICE_ROLE_KEY') ?? '' 
      : Deno.env.get('SUPABASE_ANON_KEY') ?? ''
  )
}
